tools/prerelease-tool/main.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+	"time"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+type prereleaseTool struct {
+	Data        []*github.RepositoryRelease
+	Limit       int
+	Owner       string
+	Repo        string
+	client      *github.Client
+	Prereleases []github.RepositoryRelease
+	DryRun      bool
+}
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
Do we ever see a scenario when this will take more than 30 seconds? If so the tool should complain loudly that it failed to run.

Reply:
yeah, it might actually, I meant to set that higher, to 5 minutes. there's a good amount of releases to delete now. 
I'll set a receiver on the context cancel channel to make sure I log that. great suggestion!